### PR TITLE
[NEW] New view to reactions in a message

### LINKED ISSUE #1877: [FEATURE] Possibility to see everybody who reacted to a message.
- Your Rocket.Chat app version: 4.4.0
- Your Rocket.Chat server version: 3.0.0
- Device (or Simulator) you're running with: Samsung Galaxy J7

Currently it's only possible to see 3 people of all those who reacted to a message/post, the rest won't be shown - would be great to have a full list available. 

Some examples:
WhatsApp Image 2020-04-27 at 16 51 05

WhatsApp Image 2020-04-27 at 16 46 48

### LINKED ISSUE #2152: Reactions show usernames instead of real names
- Your Rocket.Chat Experimental app version: 4.7.0.17188
- Your Rocket.Chat server version: 3.2.2
- Device (or Simulator) you're running with: Samsung Galaxy Note 8

The setting `UI_use_real_name` is ignored in the app, usernames are always shown.

### PR DESCRIPTION
**Pull request by EzequielDeOliveira**
ezgif com-video-to-gif(30)

ezgif com-video-to-gif(31)

Closes #1877 
Closes #2152

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -196,12 +196,14 @@ export default class Root extends React.Component {
 								}}
 							>
 								<ActionSheetProvider>
-									<AppContainer />
-									<TwoFactor />
-									<ScreenLockedView />
-									<ChangePasscodeView />
-									<InAppNotification />
-									<Toast />
+									<ReactionSheetProvider>
```
Comment (diegolmello) [inline app/index.js:199]: We actually need to reuse `ActionSheetProvider` here, since it's a generic.

File: app/containers/ActionSheet/Item.js [hunk 2]
```diff
@@ -6,25 +6,40 @@ import { themes } from '../../constants/colors';
...
 
 	return (
 		<Button
-			onPress={onPress}
+			onPress={reactionsMode ? () => {} : onPress}
 			style={[styles.item, { backgroundColor: themes[theme].focusedBackground }]}
 			theme={theme}
 		>
-			<CustomIcon name={item.icon} size={20} color={item.danger ? themes[theme].dangerColor : themes[theme].bodyText} />
+			{reactionsMode
```
Comment (diegolmello) [inline app/containers/ActionSheet/Item.js:25]: IMO the Item should know what to render without having to explicitly say "reaction mode".
For example:
```
{icon ? icon : null}
{avatar ? avatar : null}
```

**Comment by aKn1ghtOut** [comment]
@diegolmello if this is abandoned, should I start a fresh PR?

**Comment by diegolmello** [comment]
> @diegolmello if this is abandoned, should I start a fresh PR?

@aKn1ghtOut I'm not sure.
This PR does a lot of things, but still misses a deep refactor on our ActionSheet lib.
The idea behind this refactor is to be able to use that ActionSheet as a container for generic code.
We would use it for Reaction Picker, Message's Reactions, Message Actions, etc, but in order to do that, the code should be well structured.
That's why I prefer to develop this new component in-house, so we can discuss about pros vs cons beforehand.

You can still work on fixes for #2152 and #1877 though.

**Comment by aKn1ghtOut** [comment]
Makes sense. I'll let this be. I'll work on the other ones. Thanks a lot!

**Comment by diegolmello** [comment]
@aKn1ghtOut I was thinking about this.
I think we can make a mid term approach and fix those issues.
Forget about the whole ActionSheet abstraction I've mentioned and open a react-navigation's modal like this:
image

Can you work on it?
Thanks!

**Comment by aKn1ghtOut** [comment]
> @aKn1ghtOut I was thinking about this.
> I think we can make a mid term approach and fix those issues.
> Forget about the whole ActionSheet abstraction I've mentioned and open a react-navigation's modal like this:
> image
> 
> Can you work on it?
> Thanks!

Absolutely. Looks great too. I'll start on this as soon as possible!
Thanks!

**Pull request closed without merging**